src/query/plan/operator.cpp

```diff
@@ -6063,7 +6095,50 @@ bool PeriodicSubquery::PeriodicSubqueryCursor::Pull(Frame &frame, ExecutionConte
   OOMExceptionEnabler oom_exception;
   SCOPED_PROFILE_OP("PeriodicSubquery");
 
-  throw utils::NotYetImplemented("periodic commit");
+  if (commit_frequency_ == -1) {
+    ExpressionEvaluator evaluator(&frame, context.symbol_table, context.evaluation_context, context.db_accessor,
+                                  storage::View::OLD);
+    TypedValue commit_frequency = self_.commit_frequency_->Accept(evaluator);
+    if (commit_frequency.type() != TypedValue::Type::Int)
+      throw QueryRuntimeException("Limit on number of returned elements must be an integer.");
```

Comment:
sure, makes sense!

Reply:
however, I abstracted this into a function so it doesn't repeat over the 2 operators. I can't make the same check now since this is a public function. Are we okay to let this be like this?